### PR TITLE
CP-38617: expose xentoollog library the same way xen does

### PR DESCRIPTION
**Pull request by psafont**
This unfortunately means creating a new package that will need to be added to xs-opam so xen-api picks it as well.

This is needed to make this branch to compile in a dev environment: https://github.com/psafont/xen-api/tree/private/paus/xen-foreign

### CODE REVIEW

File: xentoollog.opam [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+bug-reports: "https://github.com/xapi-project/xenctrl/issues"
+depends: [
+  "ocaml"
+  "dune" {build}
+  "base-unix"
+]
+build: ["dune" "build" "-p" name "-j" jobs]
+depexts: [
+  ["m4" "libxen-dev" "libsystemd-dev"] {os-distribution = "debian"}
+  ["libxen-dev" "uuid-dev"] {os-distribution = "ubuntu"}
```
Comment (edwintorok) [inline xentoollog.opam]: do we really need these? I thought we just expose a mock environment here that doesn't use the real Xen. Can we drop the depexts? (the version there is going to be a lot older than what we use anyway and likely conflict)

Comment (psafont) [reply xentoollog.opam]: I think it still needs the headers to compile the stubs:

```h
#include <xentoollog.h>

#include "caml_xentoollog.h"
```
https://github.com/xapi-project/xenctrl/blob/master/libs/xentoollog/xentoollog_stubs.c#L30-L32

AFAIK these are stable bindings, so updating these shouldn't break LTS releases.

On top of that all distributions should have easy access to xen headers

Comment (psafont) [reply xentoollog.opam]: I missed these files are included here, changing

Comment (psafont) [reply xentoollog.opam]: Note that xentrl has used the depexts (as-is or through conf-xen) for years. That should get a separate review

**Review by edwintorok — APPROVED** [review]

**Review by robhoes — APPROVED** [review]

**Pull request merged**